### PR TITLE
Cache and use fprop stats in cudnn batchnorm bprop [Pt. 2]

### PR DESCRIPTION
**Pull request by csullivan**
Update to the new batch norm API introduced in #1828.

### CODE REVIEW

**Comment by csullivan** [comment]
Updated #1841 to match newly merged batch norm API and so this PR is defunct.
